src/Runner.Worker/ContainerOperationProvider.cs

```diff
@@ -494,31 +493,14 @@ private void ContainerRegistryLogout(string configLocation)
 
         private void UpdateRegistryAuthForGitHubToken(IExecutionContext executionContext, ContainerInfo container)
         {
-            var registryIsTokenCompatible = container.RegistryServer.Equals("docker.pkg.github.com", StringComparison.OrdinalIgnoreCase);
+            var registryIsTokenCompatible = container.RegistryServer.Equals("ghcr.io", StringComparison.OrdinalIgnoreCase);
             if (!registryIsTokenCompatible)
             {
                 return;
             }
 
-            var registryMatchesWorkflow = false;
-
-            // REGISTRY/OWNER/REPO/IMAGE[:TAG]
-            var imageParts = container.ContainerImage.Split('/');
-            if (imageParts.Length != 4)
-            {
-                executionContext.Warning($"Could not identify owner and repo for container image {container.ContainerImage}. Skipping automatic token auth");
-                return;
-            }
-            var owner = imageParts[1];
-            var repo = imageParts[2];
-            var nwo = $"{owner}/{repo}";
-            if (nwo.Equals(executionContext.GetGitHubContext("repository"), StringComparison.OrdinalIgnoreCase))
-            {
-                registryMatchesWorkflow = true;
```

Comment:
so the `GITHUB_TOKEN` can auth with any image from any repo?

Reply:
We want to use containers in any repo in the same org and eventually in other orgs too. So I guess we cannot do any validation here and users will just get an error if the GITHUB_TOKEN doesn't have permissions.

Reply:
@TingluoHuang We have fine grained permissions allowing/disallowing access to containers scoped to individual repositories, so while this looks more open than previously, we have policies in authzd ensuring that access is restricted as needed when pulling actual images.

Reply:
👍